tests/lib/at_cmd_parser/at_params/src/main.c

```diff
@@ -0,0 +1,448 @@
+
+	zassert_equal(TEST_PARAMS, test_list.param_count,
+		      "Params count should be the same as TEST_PARAMS");
+
+	at_params_list_free(&test_list);
+
+	zassert_equal(0, test_list.param_count,
+		      "Params list count is not 0 after free");
+	zassert_equal_ptr(NULL, test_list.params,
+			  "Params is not NULL after free");
+}
+
+static void test_params_put_get_int_setup(void)
+{
+	at_params_list_init(&test_list, TEST_PARAMS);
+}
+
+static void test_params_put_get_int(void)
+{
+	u16_t tmp_short = 0;
+	u32_t tmp_int   = 0;
+
+	zassert_equal(-EINVAL, at_params_short_put(NULL, 0, 1),
+		      "at_params_short_put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_int_put(NULL, 0, 1),
+		      "at_params_int_put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_short_put(NULL, TEST_PARAMS, 1),
+		      "at_params_short_put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_int_put(NULL, TEST_PARAMS, 1),
+		      "at_params_int_put should return -EINVAL");
+
+	zassert_equal(0, at_params_short_put(&test_list, 0, 65535),
+		      "at_params_short_put should return 0");
+	zassert_equal(0, at_params_int_put(&test_list, 1, 65536),
+		      "at_params_int_put should return 0");
+
+	zassert_equal(-EINVAL, at_params_short_get(&test_list, 1, &tmp_short),
+		      "at_params_short_get should return -EINVAL");
+
+	zassert_equal(0, at_params_int_get(&test_list, 0, &tmp_int),
+		      "at_params_int_get should return 0");
+
+	zassert_equal(0, at_params_short_get(&test_list, 0, &tmp_short),
+		      "at_params_short_get should return 0");
+	zassert_equal(65535, tmp_short, "Short should be 65535");
+
+	zassert_equal(0, at_params_int_get(&test_list, 1, &tmp_int),
+		      "at_params_int_get should return 0");
+	zassert_equal(65536, tmp_int, "Integer should be 65536");
+}
+
+static void test_params_put_get_int_teardown(void)
+{
+	at_params_list_free(&test_list);
+}
+
+static void test_params_put_get_string_setup(void)
+{
+	at_params_list_init(&test_list, TEST_PARAMS);
+	at_params_int_put(&test_list, 0, 1);
+}
+
+static void test_params_put_get_string(void)
+{
+	const char test_str[] = "Test, 1, 2, 3";
+	char test_buf[32];
+	size_t test_buf_len = sizeof(test_buf);
+
+	zassert_equal(-EINVAL, at_params_string_put(NULL, 1,
+					test_str, sizeof(test_str)),
+		      "String put should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_put(&test_list, 1,
+					NULL, sizeof(test_str)),
+		      "String put should return -EINVAL");
+
+	zassert_equal(0, at_params_string_put(&test_list, 1,
+					      test_str, sizeof(test_str)),
+		      "String put should return 0");
+
+	zassert_equal(-EINVAL, at_params_string_get(NULL, 1,
+						    test_buf, &test_buf_len),
+		      "String get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_get(&test_list, 1,
+						    NULL, &test_buf_len),
+		      "String get should return -EINVAL");
+
+	zassert_equal(-EINVAL, at_params_string_get(&test_list, 0,
```

Comment:
You could add a test to check when the output parameters( value, len) are NULL.

Reply:
I am not sure in which situation that will happen?

Reply:
To make this unit test, should you not check all possible parameter values?
Null check if the code were missing, this is why you could add a test.